ControlSystemStudio/cs-studio
# Modified archive engine to also serve JSON

See https://github.com/ControlSystemStudio/cs-studio/issues/2265.

Original message asked for me not to use no external libraries, I've used org.apache.commons.lang3 as it was used in other parts of cs-studio but can remove if needs be.

To test:
* Start the archive engine as usual
* Navigate to one of the archive engine webpages adding the format=json variable (e.g. http://localhost:4812/groups?format=json) and confirm that valid json is returned with expected data
* Navigate to the same page with format=html or no format specified and confirm that the usual webpage is shown

Note that I only converted the following pages:
* main
* group
* groups
* disconnected
* channel

as others are mainly used for control rather than readback. However, it should be easy to convert others too if required.
